frontend/public/module/k8s/k8s-models.ts

```diff
@@ -27,7 +27,7 @@ export const modelsToMap = (models: K8sKind[]): ImmutableMap<K8sResourceKindRefe
 let k8sModels = modelsToMap(_.values(staticModels));
 
 const hasModel = (model: K8sKind) =>
-  k8sModels.has(referenceForModel(model)) || k8sModels.has(model.kind);
+  model.crd ? k8sModels.has(referenceForModel(model)) : k8sModels.has(model.kind);
```

Comment:
@spadgett I have updated this code, previously knative `Service` was not registered in redux store  due to the conflicting model kind (Service).
cc: @christianvogt

Reply:
cc @andrewballantyne 
remember this piece of code? lol
What happened last time you tried to _fix_ this?

Reply:
Ah, the model collision... I had the opportunity to just provide the specific model to the badge instead of letting it reverse engineer it out of the model map using the kind.

This problem still exists in some cases (this one I assume) and https://jira.coreos.com/browse/ODC-2125

Reply:
On the flipside, it did blow out an e2e test when I broke things. PR #2917

Reply:
@andrewballantyne Yes, this issue was due to model collision. I am trying to pass classname from the model, then i realized the model itself is not registered. So i have added the above piece of code to pick appropriate models.

Reply:
I am unsure what the best call to action here is. This code seems super finicky and not really up-to-date with what we are doing with the models. I guess we wait for @spadgett and do some testing to make sure we didn't break something the e2e tests don't cover.

I know when I pushed the models together and got Knative models part of the `allModels()` map it broke the fetching of the Networking `Service` model. I don't believe the code is in any race condition, so I guess if we can still access the collision-based models as well as the Knative ones, then we may be good. Recommend you checking ODC-2125 when you do this change - if you're fully successful, that should just resolve.

Reply:
I do not trust changing this code right now. As @andrewballantyne  pointed out it's not a simple fix.
Let's address the model lookup for the color and fallback to what is currently done via className.
For knative service, we unfortunately live with it being green for 4.3. As long as we are consistently green.
Create a followup issue to track the knative service color so we can address it later.

Reply:
Sure, i will make the suggested changes